server/dash/pages/nc_summary_comparison.py

```diff
@@ -104,11 +358,325 @@ def generate_dynamic_filter(selected_nc):
         neigbhorhood council.
     """
     # If no neighborhood council is selected, use all data.
-    print("got new selected_nc ", selected_nc)
     if not selected_nc:
         df = api_data_df
     else:
         df = api_data_df[api_data_df["councilName"] == selected_nc]
 
     req_types = sorted(list(set(df["typeName"])))
     return req_types, ' '
+
+# TODO: Add request type filter if necessary? 
+@callback(
+    Output("nc_avg_comp_line_chart", "figure"),
+    [Input("selected_nc", "value")]
+)
+def update_line_chart(selected_nc):
+    """Generates a line chart visualizations for LA 311 requests data
+     based on the two selected neighborhood conucils.
+    This function takes the selected neighborhood council (nc) value
+     from the "selected_nc" dropdown and output a line chart showing 
+    the number of requests throughout the day and the average number 
+    of requests throughout the day (total number of requests / all 99 neighborhood councils).
+    Args:
+        selected_nc: A string argument automatically detected by Dash 
+        callback function when "selected_nc" element is selected in the layout.
+    Returns: 
+        nc_avg_comp_line_chart: line chart showing the number of requests
+         throughout the day for the selected neighborhood council and average
+    """
+    # If dropdown value is empty, use all data available.
+    if not selected_nc:
+        df = report_json
+        selected_nc = 'Total'
+    else:
+        df = report_json[report_json.council_name == selected_nc]
+
+    # Calculating the average number of requests throughout the day.
+    neighborhood_sum_df = df.groupby(["created_date"]).agg("sum").reset_index()  # noqa
```

Comment:
can we converge on the naming convention for the df? I see that we use "createdDate" and "created_date". I prefer the latter since Python uses that naming convention.

Reply:
so createdDate is actually the name of the dataframe from the API (api_data_df), while created_date is for report_json. should I go upstream and change that?